OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/PassphraseCacheService.java

```diff
@@ -314,7 +314,8 @@ public void onReceive(Context context, Intent intent) {
     private static PendingIntent buildIntent(Context context, long keyId) {
         Intent intent = new Intent(BROADCAST_ACTION_PASSPHRASE_CACHE_SERVICE);
         intent.putExtra(EXTRA_KEY_ID, keyId);
-        return PendingIntent.getBroadcast(context, REQUEST_ID, intent,
+        Long _id = System.currentTimeMillis();
+        return PendingIntent.getBroadcast(context, _id.intValue(), intent,
```

Comment:
Does this fix the problem? I don't see how the requestId parameter you changed is related, could you please explain?